Feature/flexible build

Ragarnoy opened the pull request:
Adds cargo make building

Ragarnoy committed:
Modify shell script, try out cargo-make. create requirements file.

Ragarnoy committed:
Remove Ugly Ass Font "Wasabi"

Ragarnoy committed:
Fix dynamic lib name for windows

Ragarnoy committed:
Environment variables for multiplatforming.

Ragarnoy committed:
Add dependency check/install for pip

Ragarnoy committed:
Cleaning build and overriding default.

arlaine4 reviewed (approved):
I tried it, it seems to work just fine

Ragarnoy committed:
Merge remote-tracking branch 'origin/feature/flexible_build' into feature/flexible_build

# Conflicts:
#	Makefile.toml

Ragarnoy committed:
Fix backslash to slash for windows in makefile

Ragarnoy committed:
Update Makefile.toml

somehow a capitalized E slipped it

Ragarnoy committed:
Unix compatibility wip

Ragarnoy committed:
Should work on xubuntu

demonoidv commented on start_gomoku.sh:
```diff
@@ -36,29 +38,37 @@ PYTHON=src/PyInterface/gomoku.py
 # Check if build is successful
-if [ $? -ne 0 ]; then
-    echo -e "\033[0;31mMaturin failed to build the project! Will NOT run the project\033[0m"
+if ! maturin develop --release; then
+    echo -e "\033[0;31mMaturin failed to build the project in a venv.\033[0m Will try to build without a virtual environment"
+    else
+      HAS_VENV=true
+    if ! pip3 install -r requirements.txt; then
+    echo -e "\033[0;31mCould not install requirements. Cannot build project\033[0m"
     exit 42
+    else
+      HAS_VENV=false
+    fi
```
Maybe an indent issue here (the first `else` should not be there)

Ragarnoy replied:
Fixed

demonoidv replied:
That's a really nice fix 😂

demonoidv reviewed (approved):
LGTM! Awesome work, it's way easier to build the product now, thank you very much for that 👍 

I only noticed a little indentation issue in the shell script, you can fix it if you want or let it as is, both work for me.

Ragarnoy committed:
Delete start_gomoku.sh

Not needed anymore since we have cargo-make. If cargo-make doesn't work we'll just have to do everything manually.

demonoidv commented:
Still good for me, feel free to merge whenever you want 👍
